torch/optim/rmsprop.py

```diff
@@ -49,7 +67,20 @@ def step(self, closure=None):
                     grad = grad.add(group['weight_decay'], p.data)
 
                 square_avg.mul_(alpha).addcmul_(1 - alpha, grad, grad)
-                avg = square_avg.sqrt().add_(group['eps'])
-                p.data.addcdiv_(-group['lr'], grad, avg)
+
+                avg = None
+                if group['centered']:
+                    grad_avg = state['grad_avg']
+                    grad_avg.mul_(alpha).add_(1 - alpha, grad)
+                    avg = square_avg.addcmul(-1, grad_avg, grad_avg).sqrt().add_(group['eps'])
+                else:
+                    avg = square_avg.sqrt().add_(group['eps'])
+
+                if group['momentum'] > 0:
+                    mom = state['momentum']
+                    mom.mul_(group['momentum']).addcdiv_(group['lr'], grad, avg)
```

Comment:
Yeah it's equivalent, but we decided that it's best to keep it consistent. The alternative formulation has the advantage that if you modify the LR, it is immediately reflected in the updates, instead of slowly making the buffer smaller.